example-application/test/global-setup.js

```diff
@@ -25,12 +24,9 @@ module.exports = async () => {
     );
 
     // ️️️✅ Best Practice: Use npm script for data seeding and migrations
-    const npmLoadAsPromise = util.promisify(npm.load);
-    await npmLoadAsPromise();
-    const npmCommandAsPromise = util.promisify(npm.commands.run);
-    await npmCommandAsPromise(['db:migrate']);
+    execSync('npm run db:migrate');
```

Comment:
I think it is better to run it asynchronously rather than synchronously, running a task that possibly takes time synchronically it's not a good practice.

Reply:
Thanks!
we block nothing here.. global setup runs in a different process.
Also, `execSync` produce better logs than `await promisifedExec` with less code:
```
➜  nodejs-integration-tests-best-practices git:(Michael/remove-npm-dep) ✗ npm t

> integration-test-a-z@1.0.0 test
> jest

Determining test suites to run...Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
Building with native build. Learn about native build in Compose here: https://docs.docker.com/go/compose-native-build/
Creating postgres-for-testing ...
Creating postgres-for-testing ... done
npm ERR! Missing script: "db:migrate1"
npm ERR!
npm ERR! Did you mean this?
npm ERR!     npm run db:migrate # run the "db:migrate" package script
npm ERR!
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/michael/.npm/_logs/2021-12-17T12_48_13_663Z-debug.log
Error: Command failed: npm run db:migrate1
npm ERR! Missing script: "db:migrate1"
npm ERR!
npm ERR! Did you mean this?
npm ERR!     npm run db:migrate # run the "db:migrate" package script
npm ERR!
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/michael/.npm/_logs/2021-12-17T12_48_13_663Z-debug.log

    at checkExecSyncError (node:child_process:826:11)
    at execSync (node:child_process:900:15)
    at module.exports (***/nodejs-integration-tests-best-practices/example-application/test/global-setup.js:26:5)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ***/nodejs-integration-tests-best-practices/node_modules/@jest/core/build/runGlobalHook.js:82:11
    at async waitForPromiseWithCleanup (***/nodejs-integration-tests-best-practices/node_modules/@jest/transform/build/ScriptTransformer.js:207:5)
    at async ***/nodejs-integration-tests-best-practices/node_modules/@jest/core/build/runGlobalHook.js:72:9
    at async pEachSeries (***/nodejs-integration-tests-best-practices/node_modules/p-each-series/index.js:8:23)
    at async _default (***/nodejs-integration-tests-best-practices/node_modules/@jest/core/build/runGlobalHook.js:58:5)
    at async runJest (***/nodejs-integration-tests-best-practices/node_modules/@jest/core/build/runJest.js:349:5)
```

VS
```
➜  nodejs-integration-tests-best-practices git:(Michael/remove-npm-dep) ✗ npm t

> integration-test-a-z@1.0.0 test
> jest

Determining test suites to run...Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
Building with native build. Learn about native build in Compose here: https://docs.docker.com/go/compose-native-build/
Creating postgres-for-testing ...
Creating postgres-for-testing ... done
Error: Command failed: npm run db:migrate1
npm ERR! Missing script: "db:migrate1"
npm ERR!
npm ERR! Did you mean this?
npm ERR!     npm run db:migrate # run the "db:migrate" package script
npm ERR!
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/michael/.npm/_logs/2021-12-17T12_46_43_384Z-debug.log

    at ChildProcess.exithandler (node:child_process:397:12)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
```

Reply:
I understand, out of interest which node version you used, I think stack trace in async errors got better